nmapscanner.py

```diff
@@ -101,43 +105,41 @@ def generate_nmap_cmd(
             if nmap_proto == "-sU":
                 protocol = "UDP"
 
-            output_logfile = output_path / f"{str(ipaddr)}_{protocol}"
+            output_logfile = output_path / sanitize_filename(f"{str(ipnet)}_{protocol}")
             # -Pn stops ping probe detection
             # -oX gives us XML output to parse
-            nmap_cmd.extend(
-                ["-Pn", nmap_proto, "-oX", str(output_logfile), str(ipaddr)]
-            )
+            nmap_cmd.extend(["-Pn", nmap_proto, "-oX", str(output_logfile), str(ipnet)])
             nmap_cmds.append(nmap_cmd)
 
     return nmap_cmds
 
 
 def nmap_prefix(
-    ipaddr: Union[IPv4Address, IPv6Address],
+    ipnet: Union[IPv4Network, IPv6Network],
     output_path: Path,
     nmap: Path,
     timeout: int,
     custom_args: List[str],
     all_ports: bool,
 ) -> int:
     nmap_cmds = generate_nmap_cmd(
-        ipaddr, output_path, nmap, timeout, custom_args, all_ports
+        ipnet, output_path, nmap, timeout, custom_args, all_ports
     )
     custom = " CUSTOM " if custom_args else " "
     for nmap_cmd in nmap_cmds:
         err = 0
         start_time = time()
         friendly_nmap_cmd = " ".join(nmap_cmd)
-        LOG.info(f"{ipaddr} -{custom}'{friendly_nmap_cmd}' starting")
+        LOG.info(f"{ipnet} -{custom}'{friendly_nmap_cmd}' starting")
         try:
-            run(nmap_cmd, stdout=PIPE, stderr=PIPE, timeout=timeout)
+            LOG.debug(run(nmap_cmd, stdout=PIPE, stderr=PIPE, timeout=timeout))
```

Comment:
You just want to dump the run completed process when debugging? makes sense I guess.

Reply:
I think long term we should stdout to /dev/null and debug log stderr I didn't do that here as I saw there were windows test so I need to work out the right way to handle that. 

There is a lot of output if we have a lot of hosts which I think will hurt us memory wise and we write to xml anyway so it is redundant. I don't see a quiet option in nmap either so that would work as well if it existed. Main reason for this is UDP and TCP scans require root or CAP_NET (I think) so if running unprivledged everything fails silently as we don't check the exit code. I have an issue open[1] in my for for that and will hopefully clean this all up when I do that.

1. https://github.com/jpaussa/nmapscanner/issues/2